optaweb-employee-rostering-backend/src/test/java/org/optaweb/employeerostering/employee/EmployeeServiceTest.java

```diff
@@ -763,4 +774,82 @@ public void updateNonExistentEmployeeAvailabilityTest() throws Exception {
+                new Employee(TENANT_ID, "Dan King", defaultContract, Collections.emptySet(), defaultCovidRisk),
+                new Employee(TENANT_ID, "Elsa Li", defaultContract, getSkillSet("Nurse"), defaultCovidRisk),
+                new Employee(TENANT_ID, "Flo Poe", defaultContract, getSkillSet("Doctor"), defaultCovidRisk),
+                new Employee(TENANT_ID, "Gus Rye", defaultContract, getSkillSet("Nurse"), defaultCovidRisk),
+                new Employee(TENANT_ID, "Hugo Smith", defaultContract, getSkillSet("Nurse"), defaultCovidRisk),
+                new Employee(TENANT_ID, "Ivy Watt", defaultContract, getSkillSet("Nur se"), defaultCovidRisk),
+                new Employee(TENANT_ID, "Jay Cole", defaultContract, getSkillSet("Nurse"), defaultCovidRisk),
+                new Employee(TENANT_ID, "Amy Green", defaultContract, getSkillSet("Nurse"), defaultCovidRisk),
+                new Employee(TENANT_ID, "Beth Jones", defaultContract,
+                             getSkillSet("Respiratory Specialist"), defaultCovidRisk),
+                new Employee(TENANT_ID, "Chad King", defaultContract, getSkillSet("Nurse"), defaultCovidRisk),
+                new Employee(TENANT_ID, "Dan Li", defaultContract, getSkillSet("Doctor"), defaultCovidRisk),
+                new Employee(TENANT_ID, "My Name", defaultContract, getSkillSet("Nur se"), defaultCovidRisk));
+
+        Function<List<Employee>, Consumer<Employee>> assertListMatch = employeeList -> expected -> {
+            Optional<Employee> maybeActual = employeeList.stream()
+                    .filter(e -> e.getName().equals(expected.getName())).findAny();
+            if (maybeActual.isPresent()) {
+                Employee actual = maybeActual.get();
+                assertEquals("Wrong contract for " + expected.getName(),
+                             expected.getContract(), actual.getContract());
+                assertEquals("Wrong covid risk type for " + expected.getName(),
+                             expected.getCovidRiskType(), actual.getCovidRiskType());
+                assertEquals("Wrong tenant id for " + expected.getName(),
+                             expected.getTenantId(), actual.getTenantId());
+                assertEquals("Wrong number of skills for " + expected.getName(),
+                             expected.getSkillProficiencySet().size(), actual.getSkillProficiencySet().size());
+                expected.getSkillProficiencySet().forEach(skill -> assertTrue("Missing skill " + skill.getName() +
+                                                                                      " for employee " + 
+                                                                                      expected.getName(),
+                                                                              actual.getSkillProficiencySet().stream()
+                                                                                      .filter(s -> s.getName()
```

Comment:
Can replace `filter().findAny().isPresent()` with `anyMatch()`.

I also recommend replacing JUnit assertions with AssertJ. With AssertJ this can be written as:
```java
assertThat(actual.getSkillProficiencySet())
        .as("Unexpected skill set for employee %s", expected.getName())
        .extracting(Skill::getName)
        .containsExactlyInAnyOrderElementsOf(expected.getSkillProficiencySet().stream()
                .map(Skill::getName)
                .collect(Collectors.toSet()));
```
which gives you additional info describing the difference between the actual and expected skill sets:
```
java.lang.AssertionError: [Unexpected skill set for employee My Name] 
Expecting:
  <["Nur se"]>
to contain exactly in any order:
  <["Nurse"]>
elements not found:
  <["Nurse"]>
and elements not expected:
  <["Nur se"]>
```

Reply:
I don't think we have AssertJ as a test dependency; I do prefer AssertJ tests as they are easier to read + write.

Reply:
We do. The AssertJ dependency is brought by `spring-boot-starter-test`. AssertJ assertions are already being used in this project.

Reply:
Odd, Eclipse doesn't show any import recommendations when I typed in `assertThat`.